Application.mk

```diff
@@ -93,6 +93,35 @@ ifneq ($(BUILD_MODULE),y)
   OBJS += $(MAINCOBJ) $(MAINCXXOBJ) $(MAINRUSTOBJ) $(MAINZIGOBJ)
 endif
 
+ifneq ($(PROGNAME),)
+  PROGOBJ := $(MAINCOBJ) $(MAINCXXOBJ) $(MAINRUSTOBJ)
+  ifneq ($(words $(PROGOBJ)), $(words $(PROGNAME)))
+    $(warning "program names $(PROGNAME) does not match mainsrcs $(PROGOBJ)")
+  endif
+  PROGLIST := $(addprefix $(BINDIR)$(DELIM),$(PROGNAME))
+  REGLIST := $(addprefix $(BUILTIN_REGISTRY)$(DELIM),$(addsuffix .bdat,$(PROGNAME)))
+
+  NLIST := $(shell seq 1 $(words $(PROGNAME)))
+  $(foreach IDX, $(NLIST), \
+    $(eval PROGKEY_$(word $(IDX),$(PROGOBJ)) := $(word $(IDX),$(PROGNAME))) \
```

Comment:
let's use the same variable prefix:
PROGKEY_->PROGNAME_
ELFPROGKEY_->PROGOBJ_
REGAPPKEY_->PROGNAME_
REGPRIORITYKEY_->PRIORITY_
REGSTACKSIZEKEY_->STACKSIZE_
REGUIDKEY_->UID_
REGGIDKEY_->GID_
REGMODEKEY_->MODE_

Reply:
> let's use the same variable prefix: PROGKEY_->PROGNAME_ ELFPROGKEY_->PROGOBJ_ REGAPPKEY_->PROGNAME_ REGPRIORITYKEY_->PRIORITY_ REGSTACKSIZEKEY_->STACKSIZE_ REGUIDKEY_->UID_ REGGIDKEY_->GID_ REGMODEKEY_->MODE_

Done✅